src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs

```diff
@@ -0,0 +1,398 @@
+                    {
+                        throw new InvalidOperationException(StringUtil.Loc("CannotDownloadFromCurrentEnvironment", hostType ?? string.Empty));
+                    }
+                    else
+                    {
+                        // This should not happen since the build id comes from build environment. But a user may override that so we must be careful.
+                        throw new ArgumentException(StringUtil.Loc("BuildIdIsNotValid", environmentBuildId));
+                    }
+                }
+
+                downloadParameters = new ArtifactDownloadParameters
+                {
+                    ProjectRetrievalOptions = BuildArtifactRetrievalOptions.RetrieveByProjectId,
+                    ProjectId = projectId,
+                    PipelineId = pipelineId,
+                    ArtifactName = artifactName,
+                    TargetDirectory = targetPath,
+                    MinimatchFilters = minimatchPatterns,
+                    MinimatchFilterWithArtifactName = true,
+                    ParallelizationLimit = int.TryParse(parallelizationLimit, out var parallelLimit) ? parallelLimit : 8,
+                    RetryDownloadCount = int.TryParse(retryDownloadCount, out var retryCount) ? retryCount : 4,
+                    CheckDownloadedFiles = bool.TryParse(checkDownloadedFiles, out var checkDownloads) && checkDownloads
+                };
+            }
+            else if (buildType == buildTypeSpecific)
+            {
+                if (String.IsNullOrEmpty(projectName))
+                {
+                    throw new ArgumentNullException(StringUtil.Loc("CannotBeNullOrEmpty"), "Project Name");
+                }
+                Guid projectId; 
+                bool isProjGuid = Guid.TryParse(projectName, out projectId);
+                if (!isProjGuid) 
+                {
+                    projectId = await GetProjectIdAsync(context, projectName);
+                }
+                // Set the default pipelineId to 0, which is an invalid build id and it has to be reassigned to a valid build id.
+                int pipelineId = 0;
+
+                if (bool.TryParse(specificBuildWithTriggering, out var specificBuildWithTriggeringBool) && specificBuildWithTriggeringBool)
+                {
+                    string hostType = context.Variables.GetValueOrDefault("system.hostType")?.Value;
+                    string triggeringPipeline = null;
+                    if (!string.IsNullOrWhiteSpace(hostType) && !hostType.Equals("build", StringComparison.OrdinalIgnoreCase)) // RM env.
+                    {
+                        var releaseAlias = context.Variables.GetValueOrDefault("release.triggeringartifact.alias")?.Value;
+                        var definitionIdTriggered = context.Variables.GetValueOrDefault("release.artifacts." + releaseAlias ?? string.Empty + ".definitionId")?.Value;
```

Comment:
trigger fix?

Reply:
Also the if and the else look like they both have definitionIdTriggered

Reply:
They do but they are from different variables